pkg/cloudmeta/gcp.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2024 ScyllaDB
+
+package cloudmeta
+
+import (
+	"context"
+	"strings"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/pkg/errors"
+)
+
+// GCPMetadata is a wrapper around gcp metadata client.
+type GCPMetadata struct {
+	meta *metadata.Client
+}
+
+// NewGCPMetadata returns gcp metadata provider.
+func NewGCPMetadata() *GCPMetadata {
+	return &GCPMetadata{
+		meta: metadata.NewClient(nil),
+	}
+}
+
+// Metadata returns InstanceMetadata from gcp if available.
+func (gcp *GCPMetadata) Metadata(ctx context.Context) (InstanceMetadata, error) {
+	machineType, err := gcp.getMachineType(ctx)
+	if err != nil {
+		return InstanceMetadata{}, errors.Wrap(err, "gcp.meta.GetWithContext")
+	}
+	return InstanceMetadata{
+		CloudProvider: CloudProviderGCP,
+		InstanceType:  machineType,
+	}, nil
+}
+
+func (gcp *GCPMetadata) getMachineType(ctx context.Context) (string, error) {
+	// The machine type for this VM. This value has the following format: projects/PROJECT_NUM/machineTypes/MACHINE_TYPE.
+	machineType, err := gcp.meta.GetWithContext(ctx, "instance/machine-type")
+	if err != nil {
+		return "", errors.Wrap(err, "gcp.meta.GetWithContext")
+	}
+
+	parts := strings.Split(machineType, "/")
+	if len(parts) < 2 {
+		return "", errors.Errorf("unexpected machine-type format: %s", machineType)
+	}
+
+	return parts[len(parts)-1], nil
```

Comment:
Could you paste a link to the docs with this format?
Why do you assert only `len(parts) < 2` instead checking for equality. Can the number of slashes be different?

Reply:
yes, you can find `machine-type` format description on this page - https://cloud.google.com/compute/docs/metadata/predefined-metadata-keys . 

I just want to extract the latest part from `/`-separated string, so firstly I make sure this string is indeed separated with `/` and secondly the latest part is extracted. This way even if number of parts will be changed for some reason this code most likely will work correctly.

Reply:
yes, you convinced me :) I'm going to update this!

Reply:
updated!